.github/workflows/ci.yml

```diff
@@ -188,45 +188,14 @@ jobs:
       - name: Stop sccache server
         run: sccache --stop-server || true
 
-      # DEBUG MODE: Upload artifacts
-      - name: Upload artifact - fluvio (DEBUG)
-        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
-        uses: actions/upload-artifact@v2
-        with:
-          name: fluvio-${{ matrix.target }}
-          path: target/${{ matrix.target }}/debug/fluvio
-      - name: Upload artifact - fluvio-run (DEBUG)
-        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
-        uses: actions/upload-artifact@v2
-        with:
-          name: fluvio-run-${{ matrix.target }}
-          path: target/${{ matrix.target }}/debug/fluvio-run
-      - name: Upload atrifact - flv-test (DEBUG)
-        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
-        uses: actions/upload-artifact@v2
-        with:
-          name: flv-test-${{ matrix.target }}
-          path: target/${{ matrix.target }}/debug/flv-test
-
-      # RELEASE MODE: Upload artifacts
-      - name: Upload artifact - fluvio (RELEASE)
-        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
-        uses: actions/upload-artifact@v2
-        with:
-          name: fluvio-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/fluvio
-      - name: Upload artifact - fluvio-run (RELEASE)
-        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
-        uses: actions/upload-artifact@v2
-        with:
-          name: fluvio-run-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/fluvio-run
-      - name: Upload atrifact - flv-test (RELEASE)
-        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
-        uses: actions/upload-artifact@v2
+      - name: Upload Fluvio Artifacts
+        uses: ./.github/actions/upload-fluvio
         with:
-          name: flv-test-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/flv-test
+          target: ${{ matrix.target }}
+          artifact: |
+            fluvio
+            fluvio-run
```

Comment:
`upload-artifact` allow multiple paths. So this could be done with that instead of customizing action?

Reply:
This action takes care of a bunch of variation that `upload-artifact` couldn't handle, such as:

- Are we in release mode or debug mode, and therefore do we upload from `target/debug` or `target/release`?
- Do we have a target other than host, so do we specify `target/debug` or `target/x86_64-unknown-linux-musl/debug`?

Basically this replaces this whole chunk from CI:

```
      # DEBUG MODE: Upload artifacts
      - name: Upload artifact - fluvio (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/fluvio
      - name: Upload artifact - fluvio-run (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-run-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/fluvio-run
      - name: Upload atrifact - flv-test (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: flv-test-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/flv-test

      # RELEASE MODE: Upload artifacts
      - name: Upload artifact - fluvio (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/fluvio
      - name: Upload artifact - fluvio-run (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-run-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/fluvio-run
      - name: Upload atrifact - flv-test (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: flv-test-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/flv-test
```